lib/path/default.nix

```diff
@@ -271,6 +271,46 @@ in /* No rec! Add dependencies on this file at the top. */ {
               second argument: "${toString path2}" with root "${toString path2Deconstructed.root}"'';
         joinRelPath components;
 
+  /*
+  Split the filesystem root from a path.
+  The result is an attribute set with these attributes:
+
+  - `root`: The filesystem root of the path, meaning that this directory has no parent directory.
```

Comment:
It's unfortunate that this concept does not have Nix documentation yet.
 - https://github.com/NixOS/nix/issues/8718
 
 (no action required now)